### PR TITLE
Dev/global refactor NEXT auth

### PR DESCRIPTION
**Pull request by TURP1**
Знімок екрана 2023-12-23 о 17 27 24
Знімок екрана 2023-12-23 о 17 27 41

### CODE REVIEW

File: src/app/(locale)/[...rest]/layout.tsx [hunk 1]
```diff
@@ -1,13 +1,12 @@
 import React, { ReactNode } from 'react';
```
Comment (AlexeYankov) [inline src/app/(locale)/[...rest]/layout.tsx:1]: delete import React

File: src/app/(locale)/sign-in/SignIn/ServiceAuth.tsx [hunk 2]
```diff
@@ -1,35 +1,56 @@
...
-          // I'll leave the then block for possible use when moving to a new translation library
-        })
-        .catch((err) => toast.error(err));
+    onSuccess: async (codeResponse) => {
+      try {
+        const data = await loginGoogleAction(codeResponse.code);
+
+        Cookies.set('accessToken', data?.data.accessToken, {
+          expires: 14,
+          sameSite: 'none',
```
Comment (AlexeYankov) [inline src/app/(locale)/sign-in/SignIn/ServiceAuth.tsx:36]: can we move it to one object and set reference on it?

File: src/utils/setCookieExpire.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+// all cookie expires after 14days
+
+export const setCookieExpires = () => {
+    return new Date(Date.now() + 14 * 24 * 60 * 60 * 1000).toUTCString()
```
Comment (AlexeYankov) [inline src/utils/setCookieExpire.ts:4]: can we move it as

const days = 24 * 60 * 60 * 1000
export const setCookieExpires = () => {
    return new Date(Date.now() + 14 * days  ).toUTCString()

**Review by AlexeYankov — APPROVED** [review]

**Review by AlexeYankov — APPROVED** [review]

**Review by maskpa100 — APPROVED** [review]

File: src/utils/cookiesActions.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  return new Date(Date.now() + 14 * days).toUTCString();
+};
+
+export const setAuthCookie = (key: string, value: string) => {
+  Cookies.set(key, value, {
+    expires: 14,
+    sameSite: 'none',
+    secure: true,
+  });
+}
```
[the remark this answers is not in the document]
Comment (maskpa100) [reply src/utils/cookiesActions.ts:17]: Снимок экрана 2023-12-25 213716

**Pull request merged**